libass/ass_render.c

```diff
@@ -2177,23 +2185,23 @@ static void render_and_combine_glyphs(ASS_Renderer *render_priv,
             }
             last_info = info;
 
-            if (!info->image || !current_info) {
-                ass_cache_dec_ref(info->image);
-                ass_cache_dec_ref(info->image_o);
+            if ((!info->bm && !info->bm_o) || !current_info) {
```

Comment:
As far as I can understand, difference is possible only in case of allocation failure. Maybe it's indeed better to skip processing altogether in case of memory problems.